perfkitbenchmarker/configs/benchmark_config_spec.py

```diff
@@ -1284,9 +1284,111 @@ def Decode(self, value, component_full_name, flag_values):
+      construct in order to decode the named option.
+    """
+    result = super(_VPNServiceSpec, cls)._GetOptionDecoderConstructions()
+    result.update({
+        'shared_key': (option_decoders.StringDecoder, {
+            'default': None,
+            'none_ok': True}),
+        'name': (option_decoders.StringDecoder, {
+            'default': None,
+            'none_ok': True}),
+        'tunnel_count': (option_decoders.IntDecoder, {
+            'default': 1,
+            'none_ok': True}),
+        'gateway_count': (option_decoders.IntDecoder, {
+            'default': 1,
+            'none_ok': True}),
+        'routing_type': (option_decoders.StringDecoder, {
+            'default': 'static',
+            'none_ok': True}),
+        'ike_version': (option_decoders.IntDecoder, {
+            'default': 1,
+            'none_ok': True}),
+    })
+    return result
+
+  @classmethod
+  def _ApplyFlags(cls, config_values, flag_values):
+    """Modifies config options based on runtime flag values.
+
+    Args:
+      config_values: dict mapping config option names to provided values. May
+          be modified by this function.
+      flag_values: flags.FlagValues. Runtime flags that may override the
+          provided config values.
+    """
+    super(_VPNServiceSpec, cls)._ApplyFlags(config_values, flag_values)
+    if flag_values['vpn_service_tunnel_count'].present:
+      config_values['tunnel_count'] = flag_values.vpn_service_tunnel_count
+    if flag_values['vpn_service_gateway_count'].present:
+      config_values['gateway_count'] = flag_values.vpn_service_gateway_count
+    if flag_values['vpn_service_name'].present:
+      config_values['name'] = flag_values.vpn_service_name
+    if flag_values['vpn_service_shared_key'].present:
+      config_values['shared_key'] = flag_values.vpn_service_shared_key
+    if flag_values['vpn_service_routing_type'].present:
+      config_values['routing_type'] = flag_values.vpn_service_routing_type
+    if flag_values['vpn_service_ike_version'].present:
+      config_values['ike_version'] = flag_values.vpn_service_ike_version
+
+
+class _VPNServiceDecoder(option_decoders.TypeVerifier):
+  """Validate the vpn_service dictionary of a benchmark config object.
+  """
+
+  def __init__(self, **kwargs):
+    super(_VPNServiceDecoder, self).__init__(valid_types=(dict,), **kwargs)
+
+  def Decode(self, value, component_full_name, flag_values):
+    """Verify vpn_service dict of a benchmark config object.
+
+    Args:
+      value: dict. Config dictionary
+      component_full_name: string.  Fully qualified name of the configurable
+        component containing the config option.
+      flag_values: flags.FlagValues.  Runtime flag values to be propagated to
+        BaseSpec constructors.
+
+    Returns:
+      _VPNService built from the config passed in in value.
+
+    Raises:
+      errors.Config.InvalidateValue upon invalid input value.
```

Comment:
Looks like the correct name is InvalidValue?  Can you fix where you copied it from as well?

Reply:
Fixed. I relied on the spark service mainly but this showed up a few places in benchmark_config_spec.